contracts/base/LendingPool.sol

```diff
@@ -238,8 +216,76 @@ contract LendingPool is Base, LendingPoolInterface {
         emit InterestWithdrawn(lender, amountToWithdraw);
     }
 
+    /**
+        @notice Update the current interest validator address.
+        @param newInterestValidator the new interest validator address.
+     */
+    function setInterestValidator(address newInterestValidator)
+        external
+        whenAllowed(msg.sender)
+    {
+        require(newInterestValidator.isContract(), "VALIDATOR_MUST_CONTRACT_NT_EMPTY");
+        address oldInterestValidator = address(interestValidator);
+        oldInterestValidator.requireNotEqualTo(
+            newInterestValidator,
+            "NEW_VALIDATOR_MUST_BE_PROVIDED"
+        );
+
+        interestValidator = InterestValidatorInterface(newInterestValidator);
+
+        emit InterestValidatorUpdated(
+            msg.sender,
+            oldInterestValidator,
+            newInterestValidator
+        );
+    }
+
+    /**
+        @notice It initializes the contract state variables.
+        @param zTokenAddress zToken token address.
+        @param lendingTokenAddress ERC20 token address.
+        @param lendersAddress Lenders contract address.
+        @param loansAddress Loans contract address.
+        @param settingsAddress Settings contract address.
+        @param marketsAddress Markets state conntract address.
+        @param interestValidatorAddress Interest validator address. It can be 0x0.
+        @dev It throws a require error if the contract is already initialized.
+     */
+    function initialize(
+        address zTokenAddress,
+        address lendingTokenAddress,
+        address lendersAddress,
+        address loansAddress,
+        address cTokenAddress,
+        address settingsAddress,
+        address marketsAddress,
+        address interestValidatorAddress
+    ) external isNotInitialized() {
+        zTokenAddress.requireNotEmpty("ZTOKEN_ADDRESS_IS_REQUIRED");
+        lendingTokenAddress.requireNotEmpty("TOKEN_ADDRESS_IS_REQUIRED");
+        lendersAddress.requireNotEmpty("LENDERS_ADDRESS_IS_REQUIRED");
+        loansAddress.requireNotEmpty("LOANS_ADDRESS_IS_REQUIRED");
+        require(
+            interestValidatorAddress.isEmpty() || interestValidatorAddress.isContract(),
+            "VAL_MUST_BE_EMPTY_OR_CONTRACT"
+        );
+
+        _initialize(settingsAddress, marketsAddress);
+
+        zToken = ZTokenInterface(zTokenAddress);
+        lendingToken = IERC20(lendingTokenAddress);
+        lenders = LendersInterface(lendersAddress);
+        loans = loansAddress;
+        cToken = cTokenAddress;
+        interestValidator = InterestValidatorInterface(interestValidatorAddress);
+    }
+
```

Comment:
Again should leave the `initializer` towards the top.

Reply:
The same reason.